lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -152,11 +188,23 @@ def format_and_sign(post)
         post
       end
 
+      def get_network_payment_reference(response)
+        parameters = { request: { MerchantId: @options[:merchant_id], OrderId: response.params['response'][:OrderId] } }
+        body = post_data format_and_sign(parameters)
+
+        raw_response = ssl_request :post, url('query'), body, {}
+        response = parse(raw_response)
+        Response.new(true, response[:response][:Payment][:NetworkPaymentReference], {})
```

Comment:
I think you'll want to use `dig` here to guard against `nil` but it also might be better to retrieve the value outside of building the `Response` object.